beacon_node/eth1/src/web3_fetcher.rs

```diff
@@ -0,0 +1,336 @@
+                    println!("Error getting block number");
+                    Error::Web3Error(e)
+                })
+                .timeout(Duration::from_secs(10))
+                .map_err(|_| {
+                    println!("Timed out getting block_number");
+                    Error::Timeout
+                }),
+        )
+    }
+
+    /// Get `deposit_count` from deposit contract at given eth1 block number.
+    fn get_deposit_count(
+        &self,
+        block_number: Option<BlockNumber>,
+    ) -> Box<dyn Future<Item = Result<u64>, Error = Error> + Send> {
+        Box::new(
+            self.contract
+                .query(
+                    "get_deposit_count",
+                    (),
+                    None,
+                    Options::default(),
+                    block_number,
+                )
+                .map(|x| {
+                    let data: Vec<u8> = x;
+                    vec_to_u64_le(&data)
+                })
+                .map_err(|e| {
+                    println!("Error getting deposit count");
+                    Error::ContractError(e)
+                })
+                .timeout(Duration::from_secs(10))
+                .map_err(|_| {
+                    println!("Timed out getting deposit count");
+                    Error::Timeout
+                }),
+        )
+    }
+
+    /// Get `deposit_root` from deposit contract at given eth1 block number.
+    fn get_deposit_root(
+        &self,
+        block_number: Option<BlockNumber>,
+    ) -> Box<dyn Future<Item = H256, Error = Error> + Send> {
+        Box::new(
+            self.contract
+                .query(
+                    "get_hash_tree_root",
+                    (),
+                    None,
+                    Options::default(),
+                    block_number,
+                )
+                .map(|x: Vec<u8>| H256::from_slice(&x))
+                .map_err(|e| {
+                    println!("Error getting deposit root");
+                    Error::ContractError(e)
+                })
+                .timeout(Duration::from_secs(10))
+                .map_err(|_| {
+                    println!("Timed out getting deposit root");
+                    Error::Timeout
+                }),
+        )
+    }
+
+    /// Get `DepositEvent` events in given range.
+    fn get_deposit_logs_in_range(
+        &self,
+        start_block: BlockNumber,
+        end_block: BlockNumber,
+        cache: Arc<RwLock<BTreeMap<u64, DepositData>>>,
+    ) -> Box<dyn Future<Item = (), Error = Error> + Send> {
+        /// Keccak256 hash of "DepositEvent" in bytes for passing to log filter.
+        const DEPOSIT_CONTRACT_HASH: &str =
+            "649bbc62d0e31342afea4e5cd82d4049e7e1ee912fc0889aa790803be39038c5";
+        let filter = FilterBuilder::default()
+            .address(vec![self.contract.address()])
+            .topics(
+                Some(vec![DEPOSIT_CONTRACT_HASH.parse().unwrap()]),
```

Comment:
I think it would be nice to have an `expect` with a message instead of an `unwrap`.

That will help find the problem if it occurs.